doc/nrf/protocols/matter/end_product/versioning.rst

```diff
@@ -0,0 +1,88 @@
+.. _ug_versioning_in_matter:
+
+Maintaining firmware version
+############################
+
+.. contents::
+   :local:
+   :depth: 2
+
+To effectively deploy a Matter product, it is essential to implement application version management. This versioning is crucial for managing firmware upgrades on devices, and it is also displayed within ecosystem applications, as provided by the Basic Information cluster.
+
+There are two primary approaches for maintaining versioning. It's important to note that these approaches should not be used simultaneously:
+
+Utilizing a :file:`VERSION` file as detailed in the Zephyr Project documentation: :ref:`Zephyr Project Versioning<app-version-details>`. This method involves defining the version information in a specific file format.
+
+Implementing dedicated Kconfig configurations. This approach uses Kconfig system configurations to set and manage the versioning details.
+
+Choose the approach that best aligns with your project requirements and infrastructure.
+
+Using :file:`VERSION` file
+**************************
+
+To implement versioning based on a :file:`VERSION` file, you must create a file named :file:`VERSION` within the application directory and ensure it contains the appropriate content:
+
+.. code-block:: console
+
+    VERSION_MAJOR =
+    VERSION_MINOR =
+    PATCHLEVEL =
+    VERSION_TWEAK =
+
+.. note::
+   You must assign a value to at least one of the variables.
+   If not, the :file:`VERSION` file will cause error.
+
+For example:
+
+.. code-block:: console
+
+   VERSION_MAJOR = 2
+   VERSION_MINOR = 5
+   PATCHLEVEL = 99
+   VERSION_TWEAK = 0
+
+A :file:`VERSION` file is responsible for assigning values to MCUboot version in the following format: ``MAJOR.MINOR.PATCH.TWEAK``.
```

Comment:
`+TWEAK` here as well? Also, below in the format you use spaces around dots and here you don't - can this be aligned?

Moreover, shouldn't we explain how Matter OTA image version is derived out of these components?